### PR TITLE
Add flake8 workflow, clean up whitespace, and remove unwanted prints

### PR DESCRIPTION
**Pull request by subramanya1997**
- Implement a GitHub Actions workflow to automatically run flake8 linting checks on pull requests. The flake8 workflow configuration can be found in the .github/workflows directory under the file named flake8.yml.
- Clean up extraneous whitespace characters from various parts of the codebase for improved code readability and style consistency.
- Delete unnecessary and debug-related print statements from the code to enhance code clarity and reduce clutter.

This is clean up effort with respect to issue #137

### CODE REVIEW

File: .github/workflows/flake8.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 --extend-ignore E501,E302,E722,F841 . --count --show-source --statistics
```
Comment (ryanpeach) [inline .github/workflows/flake8.yml]: I'd use a setup.cfg like in my PR.

File: .github/workflows/flake8.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8
```
Comment (ryanpeach) [inline .github/workflows/flake8.yml]: I'd create a `requirements-dev.txt` instead.

File: .github/workflows/flake8.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Flake8 Lint
```
Comment (ryanpeach) [inline .github/workflows/flake8.yml]: Call it "linting" in general so we can add other tools later.

File: .github/workflows/flake8.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+jobs:
+  flake8-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
```
Comment (ryanpeach) [inline .github/workflows/flake8.yml]: We are on 3.11 I believe.

**Review by ryanpeach — CHANGES_REQUESTED** [review]
Minor complaint but otherwise totally approve and appreciate.

I'd just steal the `requirements-dev.txt`, the workflow, and the setup.cfg (with your modifications) from my PR. Then keep all the code changes from your PR.

File: .github/workflows/flake8.yml (hunk 4 above)
Comment (subramanya1997) [reply .github/workflows/flake8.yml]: WIll update it.

File: .github/workflows/flake8.yml (hunk 1 above)
Comment (subramanya1997) [reply .github/workflows/flake8.yml]: will steal your code!

**Comment by subramanya1997** [comment]
@ryanpeach made the necessary changes required. Let me know if you think anything needs to be changed.

**Review by ryanpeach — APPROVED** [review]
My CICD issues are resolved.

**Review by nponeccop — CHANGES_REQUESTED** [review]
Merge the linting-related changes first. For example, create a separate PR with just whitespace correction to match flake8. When it is merged you rebase this one on top and make another small change. And finally when all linting is fixed in the master, you can all the CI to enforce it. Also note that we need to combine this with merging existing PRs so they don't become unmergeable en masse.

@ryanpeach @Torantulino what do you think?

**Comment by nponeccop** [comment]
@subramanya1997 Now there are conflicts (partially due to a big scope). Split it up and/or resolve the conflicts.

**Comment by subramanya1997** [comment]
yeah will do.

**Comment by ryanpeach** [comment]
> Merge the linting-related changes first. For example, create a separate PR with just whitespace correction to match flake8. When it is merged you rebase this one on top and make another small change. And finally when all linting is fixed in the master, you can all the CI to enforce it. Also note that we need to combine this with merging existing PRs so they don't become unmergeable en masse.
> 
> @ryanpeach @Torantulino what do you think?

I think while ideal, turning a one step processes into 3 step processes, in a repo that has such a slow PR cycle, is a bad idea. It is perfectly fine to merge a flake8 workflow, with flake8 fixes, all at once, even professionally, whereas the alternative will be long and tedious and not improve the quality of the process in any discernable way.

@nponeccop you're doing good work but you aren't a maintainer, and don't know python (by your own admission). Please preface every message/instruction as your opinion, rather than the opinion of the repo.

**Comment by ryanpeach** [comment]
It's a plan, but here's another plan:

Get all the code quality changes merged into @Torantulino's machine. Test it thoroughly over the course of a day. Become happy with it. Push it to master.

Now, you've broken everyone's PR, but:

1. You only did it once, whereas with my PR's they have been broken at least 4 times over the weekend.
2. Now you have a CICD pipeline to review everyone's changes
3. You did it overnight, and now you can work on features in a clean environment.

**Comment by richbeales** [comment]
Thanks for this PR, we've now implemented Flake8 linting.  I'll close this PR for now.

**Pull request closed without merging**